modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeUpdatedEulerAngle.C

```diff
@@ -0,0 +1,72 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeUpdatedEulerAngle.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeUpdatedEulerAngle);
+
+InputParameters
+ComputeUpdatedEulerAngle::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription(
+      "This class computes the updated Euler angle for crystal plasticity simulations. This needs "
+      "to be used together with the  ComputeMultipleCrystalPlasticityStress class, where the "
+      "updated rotation material property is computed. ");
+  params.addParam<bool>(
+      "radian_to_degree", true, "Whether to convert euler angles from radian to degree.");
+  return params;
+}
+
+ComputeUpdatedEulerAngle::ComputeUpdatedEulerAngle(const InputParameters & parameters)
+  : Material(parameters),
+    _updated_rotation(getMaterialProperty<RankTwoTensor>("updated_rotation")),
+    _updated_euler_angle(declareProperty<RealVectorValue>("updated_Euler_angle"))
+{
+}
+
+void
+ComputeUpdatedEulerAngle::initQpStatefulProperties()
+{
+  _updated_euler_angle[_qp].zero();
+}
+
+void
+ComputeUpdatedEulerAngle::computeQpProperties()
+{
+  computeEulerAngleFromRotationMatrix(_updated_rotation[_qp], _updated_euler_angle[_qp]);
+}
+
+void
+ComputeUpdatedEulerAngle::computeEulerAngleFromRotationMatrix(const RankTwoTensor & rot,
```

Comment:
Did you take a look at `EulerAngles` class? I just found that class. I am not sure if that is useful for you or not. We might want to reduce any code duplications if possible.

Reply:
Thanks Wen! 

I remember taking a look into the `EulerAngles` class some time ago. Looks like it aims at transforming the Quaternion notation to Euler angle, not transforming the rotation matrix to Euler angle. 

Eigen might be able to transform rotation matrix to Quaternion then to Euler angle though. Let me take a closer look and see if we can simplify this part of the code

Reply:
I tried to make use of the existing `EuerAngles` class. The `Eigen` library enables simpler transformation from rotation matrix to Euler angles, so I do not have to implement the transformation separately, which is good.

One thing is that I'll need to transform the rotation matrix material property (declared as RankTwoTensor in the CP base classes) to Eigen::Matrix (see below), which does not seem ideal. I do not have a good idea to address this now without changing things in the CP base classes. It maybe better to improve this while addressing #19497 if that's okay.